js/pawn.js

```diff
@@ -0,0 +1,52 @@
+import { Piece } from "./piece.js"
+
+export class Pawn extends Piece{
+
+    constructor(pieceId, rowIndex, colIndex, playerId) {
+        super()
+
+        this.id = pieceId;
+        this.rowPos = rowIndex
+        this.colPos = colIndex;
+        this.prevRowPos = 0;
+        this.prevColPos = 0;
+        this.moved = false;
+        this.playerId = playerId;
+        this.lastTake = null;
+        this.unicodeChar = "&#9817;";
+    }
+
+    possibleMoves(board) {
+        let possibleMoves = [];
+        var moveDir = 0;
+
+        if (this.playerId == 0) { // Check which direction the piece can move
+            moveDir = 1;
+        }
+        else {
+            moveDir = -1;
+        }
+        
+        if (!this.moved &&
+                board[this.rowPos+(moveDir*2)][this.colPos] === undefined &&
+                board[this.rowPos+moveDir][this.colPos] === undefined) { // Check 'in front' of the piece if it hasn't moved yet
+            possibleMoves.push([this.rowPos+(moveDir*2), this.colPos]);
+        }
+
+        if (0 <= this.rowPos+moveDir <= 7) {
```

Comment:
The statement `0 <= this.rowPos+moveDir <= 7` in each `if` statement will return incorrect results since the statement will be evaluated as `((0 <= this.rowPos+moveDir) <= 7)` and compare a boolean to a number.